src/System.Net.Sockets/tests/FunctionalTests/SendReceive.cs

```diff
@@ -226,6 +226,115 @@ public SendReceive(ITestOutputHelper output)
+
+                            Assert.Equal(3, receive1.Result + receive2.Result + receive3.Result);
+                        }
+                    }
+                }
+            }
+        }
+
+        [OuterLoop] // TODO: Issue #11345
+        [Theory]
+        [MemberData(nameof(LoopbacksAndBuffers))]
+        public async Task SendRecv_Stream_TCP_MultipleConcurrentSends(IPAddress listenAt, bool useMultipleBuffers)
+        {
+            using (var server = new Socket(listenAt.AddressFamily, SocketType.Stream, ProtocolType.Tcp))
+            {
+                server.BindToAnonymousPort(listenAt);
+                server.Listen(1);
+
+                EndPoint clientEndpoint = server.LocalEndPoint;
+                using (var client = new Socket(clientEndpoint.AddressFamily, SocketType.Stream, ProtocolType.Tcp))
+                {
+                    Task clientConnect = ConnectAsync(client, clientEndpoint);
+                    using (Socket remote = await AcceptAsync(server))
+                    {
+                        await clientConnect;
+
+                        Task<int> send1, send2, send3;
+                        if (useMultipleBuffers)
+                        {
+                            var bufferList1 = new List<ArraySegment<byte>> { new ArraySegment<byte>(new byte[1000000]), new ArraySegment<byte>(new byte[1000000]) };
+                            var bufferList2 = new List<ArraySegment<byte>> { new ArraySegment<byte>(new byte[1000000]), new ArraySegment<byte>(new byte[1000000]) };
+                            var bufferList3 = new List<ArraySegment<byte>> { new ArraySegment<byte>(new byte[1000000]) };
+
+                            send1 = SendAsync(client, bufferList1);
+                            send2 = SendAsync(client, bufferList2);
+                            send3 = SendAsync(client, bufferList3);
+                        }
+                        else
+                        {
+                            var buffer1 = new ArraySegment<byte>(new byte[2000000]);
+                            var buffer2 = new ArraySegment<byte>(new byte[2000000]);
+                            var buffer3 = new ArraySegment<byte>(new byte[1000000]);
+
+                            send1 = SendAsync(client, buffer1);
+                            send2 = SendAsync(client, buffer2);
+                            send3 = SendAsync(client, buffer3);
+                        }
+
+                        int receivedTotal = 0;
+                        int received;
+                        var receiveBuffer = new ArraySegment<byte>(new byte[1000000]);
+                        while (receivedTotal < 5000000)
+                        {
+                            if ((received = await ReceiveAsync(remote, receiveBuffer)) == 0) break;
+                            receivedTotal += received;
+                        }
+                        Assert.Equal(5000000, receivedTotal);
```

Comment:
It would be nice to test that the sends (and receives above) are processed in the order they are issued.  Maybe send the buffer # at the beginning of each and then verify these on receive?